src/backend/distributed/utils/background_jobs.c

```diff
@@ -946,6 +1033,11 @@ CitusBackgroundTaskQueueMonitorMain(Datum arg)
 	MemoryContextSwitchTo(firstContext);
 	MemoryContextDelete(backgroundTaskContext);
 
+	if (GotSigterm)
+	{
+		proc_exit(1);
+	}
```

Comment:
I think it's common for correctly handled SIGTERM to return a 0 exit code, so I don't think this code is necessary.